rust/Earthfile

```diff
@@ -32,6 +33,13 @@ INIT:
     ARG sweep_days=4
     ENV EARTHLY_SWEEP_DAYS=$sweep_days
 
+    # Make sure that crates installed through this function are stored in the original cargo home, and not in the cargo home within the mount cache.
+    # This way, if BK garbage-collects them, the build is not broken.
+    ENV CARGO_INSTALL_ROOT=$CARGO_HOME
+    # We change $CARGO_HOME while keeping $ORIGINAL_CARGO_HOME/bin directory in the path. This way, the Cargo binary is still accessible and the whole $CARGO_HOME is within the global cache
+    # ($CARGO_HOME/.package-cache has to be in the cache so Cargo can properly synchronize parallel access to $CARGO_HOME resources).
+    ENV CARGO_HOME="/tmp/earthly/.cargo"
```

Comment:
Doing this here implies that users must be careful to always call `cargo` using the mounts from `GET_RUST_CACHE_MOUNTS` in place, else they will end up with a confusing situation with `/tmp/earthly/.cargo` existing in both the mount caches and the layers.

I suppose that isn't too different to before where you could end up with `$HOME/.cargo` in the layer cache and `/tmp/earthly/.cargo` in the mount cache which was also a potential footgun.

I can't think of a technical solution to this, but maybe the README should say something about being sure you always use `DO +CARGO` or the mounts from `DO +GET_RUST_CACHE_MOUNTS` if doing "cargo stuff"?

Reply:
Makes sense Ian. I will document this